src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,139 +53,179 @@ export interface MenuProps {
+  await nextTick();
+  if (disabledKeyboardNavigation.value) {
+    menuItemsTemplateRefs.value.forEach((item) => {
+      item.tabindex = 0;
+    });
+  }
+  if (props.enableKeyboardNavigation
+      && menuItemsTemplateRefs.value.length > 0) {
+    setNegativeTabindexForNonInitialMenuItems();
+  }
+});
+onMounted(async () => {
+  await nextTick();
+  if (props.enableKeyboardNavigation
+      && menuItemsTemplateRefs.value.length > 0) {
+    setNegativeTabindexForNonInitialMenuItems();
+  }
+});
+const lastFocusedMenuItemTemplateRefs = ref<ElementRef | null>(null);
+const handleMenuItemFocus = async (element: ComputedRef<ElementRef | undefined>) => {
+  if (element.value && initialElement.value) {
+    await focusElement(element.value.itemTemplateRefs.content.$el, true);
+    initialElement.value.tabindex = -1;
+  }
+};
+const handleMenuKeyDown = async ({ key }: KeyboardEvent) => {
+  if (disabledKeyboardNavigation.value) {
+    return;
+  }
+  const activeElement = focusedElement.value;
   switch (key) {
     case 'ArrowUp':
-      if (prevMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        prevMenuItem.value.tabindex = 0;
-        focusElement(prevMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(prevElement);
       break;
     case 'ArrowDown':
-      if (nextMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        nextMenuItem.value.tabindex = 0;
-        focusElement(nextMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(nextElement);
       break;
     case 'Home':
     case 'PageUp':
-      if (firstMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        firstMenuItem.value.tabindex = 0;
-        focusElement(firstMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(firstElement);
       break;
     case 'End':
     case 'PageDown':
-      if (lastMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        lastMenuItem.value.tabindex = 0;
-        focusElement(lastMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(lastElement);
       break;
     case 'Tab':
-      const lastFocusedElement = ref(focusedMenuItem.value);
-      setTimeout(() => {
-        if(!lastFocusedElement.value) return;
-        lastFocusedElement.value.tabindex = -1;
-        if (selectedMenuItem.value) {
-          selectedMenuItem.value.tabindex = 0;
-          return;
-        }
-        if (firstMenuItem.value) {
-          firstMenuItem.value.tabindex = 0;
-        }
-      }, 0);
+      if (initialElement.value === activeElement) {
+        return;
+      }
+      if (activeElement) {
+        activeElement.tabindex = 0;
+        // We need to use setTimeout() because nextTick() does not bring the desired effect.
```

Comment:
Have you investigated why?

Reply:
yes, because it's not related to vue update. Here we need to delay this changes a little bit

Reply:
Wait, why do we need to do 2 reassignments in a row anyway?